webapp/niapp.yaml

```diff
@@ -0,0 +1,418 @@
+      description: List the emails with which the user has shared webapps
+      operationId: list-shared-emails
+      responses:
+        200:
+          $ref: '#/responses/SharedEmailsResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+  '/webapps/{id}/content':
+    get:
+      tags:
+        - content
+      summary: Get webapp content
+      description: Get the content of a webapp. ContentType varies from JSON for dashboards and templates or redirect to main HTML for WebVIs
+      operationId: get-content-index
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          description: The webapp's content. JSON for Dashboards, HTML for WebVIs
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    put:
+      tags:
+        - content
+      summary: Upload content for an existing webapp
+      description: Upload content for an existing webapp
+      operationId: update-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: body
+          required: true
+          name: content
+          description: The webapp content to upload
+          schema:
+            $ref: '#/definitions/WebAppContent'
+      responses:
+        200:
+          description: Success
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/api/webapps/{id}/content/{path}':
+    get:
+      tags:
+        - content
+      summary: Get webapp content by path
+      description: Get the webapp content at a path. This applies to webapps with multiple files, such as WebVIs.
+      operationId: get-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/WebAppContentPath'
+      responses:
+        200:
+          $ref: '#/definitions/WebAppContent'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+  '/webapps/{id}/duplicate/{sourceId}':
+    post:
+      tags:
+        - content
+      summary: Duplicates webapp content
+      description: Duplicates webapp content
+      operationId: duplicate-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: path
+          type: string
+          required: true
+          name: sourceId
+          description: The Id of the source webapp to get copy content from
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+parameters:
+  WebAppId:
+    in: path
+    name: id
+    description: The webapp identifier
+    type: string
+    required: true
+  WebAppContentPath:
+    in: path
+    name: path
+    description: Path to webapp content
+    type: string
+    required: true
+  CreateWebAppRequest:
```

Comment:
The request body for the create/update metadata routes are missing parts of the model, such as the document type.

Reply:
addendum: the "shared" and "policyIds" parameters are required to create a webapp, but this isn't documented here.

Reply:
Yeah, these parameters are currently required. But we moved updating the sharing parameters to a new route. And the shared parameters are getting removed:

https://github.com/ni-kismet/helium-webappservices/pull/374